vtm-android/src/org/oscim/android/tiling/MBTilesBitmapTileDataSource.java

```diff
@@ -0,0 +1,281 @@
+            if (cursor.moveToFirst()) {
+                return cursor.getBlob(0);
+            }
+        } finally {
+            if (cursor != null)
+                cursor.close();
+        }
+        return null;
+    }
+
+    public String getVersion() throws Exception {
+        checkMetadata();
+        return metadataMap.get("version");
+    }
+
+    private static android.graphics.Bitmap makeBitmapTransparent(android.graphics.Bitmap originalBitmap, int alpha) {
+        android.graphics.Bitmap newBitmap = android.graphics.Bitmap.createBitmap(originalBitmap.getWidth(), originalBitmap.getHeight(), android.graphics.Bitmap.Config.ARGB_8888);
+        Canvas canvas = new Canvas(newBitmap);
+        Paint alphaPaint = new Paint();
+        alphaPaint.setAlpha(alpha);
+        canvas.drawBitmap(originalBitmap, 0, 0, alphaPaint);
+        return newBitmap;
+    }
+
+    private static android.graphics.Bitmap makeTransparent(android.graphics.Bitmap bit, int colorToRemove) {
+        int width = bit.getWidth();
+        int height = bit.getHeight();
+        android.graphics.Bitmap myBitmap = android.graphics.Bitmap.createBitmap(width, height, android.graphics.Bitmap.Config.ARGB_8888);
+        int[] allpixels = new int[myBitmap.getHeight() * myBitmap.getWidth()];
+        bit.getPixels(allpixels, 0, myBitmap.getWidth(), 0, 0, myBitmap.getWidth(), myBitmap.getHeight());
+        myBitmap.setPixels(allpixels, 0, width, 0, 0, width, height);
+
+        for (int i = 0; i < myBitmap.getHeight() * myBitmap.getWidth(); i++) {
+            if (allpixels[i] == colorToRemove)
+                allpixels[i] = Color.alpha(Color.TRANSPARENT);
+        }
+
+        myBitmap.setPixels(allpixels, 0, myBitmap.getWidth(), 0, 0, myBitmap.getWidth(), myBitmap.getHeight());
+        return myBitmap;
+    }
+
+    private static int[] osmTile2TmsTile(int tx, int ty, int zoom) {
+        return new int[]{tx, (int) ((Math.pow(2, zoom) - 1) - ty)};
+    }
+
+    @Override
+    public void query(MapTile tile, ITileDataSink sink) {
+        QueryResult res = FAILED;
+
+        try {
+            byte[] imageBytes = getTile(tile.tileX, tile.tileY, tile.zoomLevel);
+            if (mTransparentColor != null || alpha != null) {
+                android.graphics.Bitmap bmp = BitmapFactory.decodeByteArray(imageBytes, 0, imageBytes.length);
+                if (mTransparentColor != null) {
+                    bmp = makeTransparent(bmp, mTransparentColor);
+                }
+                if (alpha != null) {
+                    bmp = makeBitmapTransparent(bmp, alpha);
+                }
+                ByteArrayOutputStream bos = new ByteArrayOutputStream();
+                bmp.compress(android.graphics.Bitmap.CompressFormat.PNG, 100, bos);
+                imageBytes = bos.toByteArray();
+            }
+
+            Bitmap bitmap = AndroidGraphics.decodeBitmap(new ByteArrayInputStream(imageBytes));
```

Comment:
Can use the global `CanvasAdapter.decodeBitmap` to be ready when the feature supports all platforms.